### PR TITLE
SDIT-1175-IdentifiersResource-and-IncidentsResource

### PR DESCRIPTION
**Pull request by steverendell**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/BookingMovementsResource.java [hunk 1]
```diff
@@ -69,7 +69,7 @@ public class BookingMovementsResource {
             @ApiResponse(responseCode = "400", description = "Invalid request.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class))}),
             @ApiResponse(responseCode = "404", description = "Requested resource not found.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class))}),
             @ApiResponse(responseCode = "500", description = "Unrecoverable error occurred whilst processing request.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class))})})
-    @Operation(summary = "Schedules a prison to court hearing for an offender and given court case.", description = "Schedules a prison to court hearing for an offender and given court case.")
+    @Operation(summary = "Schedules a prison to court hearing for an offender and given court case.", description = "Schedules a prison to court hearing for an offender and given court case. Requires role COURT_HEARING_MAINTAINER and scope write")
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/BookingMovementsResource.java:72]: comment from an earlier PR

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/IncidentsResource.java [hunk 2]
```diff
@@ -33,11 +34,11 @@ public class IncidentsResource {
             @ApiResponse(responseCode = "400", description = "Invalid request.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class))}),
             @ApiResponse(responseCode = "404", description = "Requested resource not found.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class))}),
             @ApiResponse(responseCode = "500", description = "Unrecoverable error occurred whilst processing request.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class))})})
-    @Operation(summary = "Return an Incident for a given incident ID", description = "System access only")
+    @Operation(summary = "Return an Incident for a given incident ID", description = "Requires the VIEW_PRISONER_DATA role.")
     @GetMapping("/{incidentId}")
+    @PreAuthorize("hasAnyRole('SYSTEM_USER', 'VIEW_PRISONER_DATA')")
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/IncidentsResource.java:39]: Added VIEW_PRISONER_DATA  which the client already has

File: src/main/java/uk/gov/justice/hmpps/prison/service/InmateService.java [hunk 3]
```diff
@@ -415,11 +415,6 @@ public List<OffenderIdentifier> getOffenderIdentifiers(final Long bookingId, @Nu
                 .toList();
     }
 
-    @VerifyBookingAccess(overrideRoles = {"SYSTEM_USER", "GLOBAL_SEARCH", "VIEW_PRISONER_DATA"})
-    public List<OffenderIdentifier> getOffenderIdentifiersByTypeAndValue(@NotNull final String identifierType, @NotNull final String identifierValue) {
-        return repository.getOffenderIdentifiersByTypeAndValue(identifierType, identifierValue);
-    }
-
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/service/InmateService.java:422]: Not used

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/IncidentsResource.java (hunk 2 above)
Comment (andymarke) [reply src/main/java/uk/gov/justice/hmpps/prison/api/resource/IncidentsResource.java:39]: do we need to add SYSTEM_USER ? (FYI @karenmillermoj )

Comment (andymarke) [reply src/main/java/uk/gov/justice/hmpps/prison/api/resource/IncidentsResource.java:39]: Actually, where is this API used? Is it Prisoner Profile or in some incident reporting ? Might be a opportunity to add a different role - this feels very incident specific rather something that should be exposed to some one that just needs core prisoner data. VIEW_INCIDENTS might make more sense, but not sure if this is getting into @karenmillermoj  territory

**Review by andymarke — APPROVED** [review]
LGTM

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/IncidentsResource.java (hunk 2 above)
Comment (karenmillermoj) [reply src/main/java/uk/gov/justice/hmpps/prison/api/resource/IncidentsResource.java:39]: I'm happy to tidy this up as part of my work @steverendell

Comment (steverendell) [reply src/main/java/uk/gov/justice/hmpps/prison/api/resource/IncidentsResource.java:39]: OK. The only client is the risk-profiler which currently has SYSTEM_USER and VIEW_PRISONER_DATA. So i could protect this with `hasAnyRole('SYSTEM_USER', 'VIEW_INCIDENTS')`. There are other incident endpoints which should be addressed too though, I could look at them next.

Comment (steverendell) [reply src/main/java/uk/gov/justice/hmpps/prison/api/resource/IncidentsResource.java:39]: Ill do this for the 2 risk-profiler endpoints and remove the others which are not called

**Review by andymarke — APPROVED** [review]
lgtm

**Pull request merged**